andela/ah-backend-poseidon
# #162162999  Send a verification link to a user via email upon successful registration.

### What does this pull request do?
- This PR verifies a user after a successful signup.
- A user verifies new account created via clicking a link provided in the email.
  
### Description of the tasks to be completed?
- When a user signs up, he should receive an email verification link.
- When the user clicks on the link provided, he should be redirected to verify account page.

### How should this be manually tested?

First clone this repository
```
$ git clone https://github.com/andela/ah-backend-poseidon
$ cd ah-backend-poseidon
```
Create virtual environment and install it
```
$ python3 -m venv venv
$ source/venv/bin/activate
```
Then install all the necessary dependencies
```
pip install -r requirements.txt
```
Run the application
At the terminal or console type
```
 $ python manage.py makemigrations
 $ python manage.py migrate
 $ python manage.py runserver
```
Head over to postman and register a user through this endpoint /api/users/. 
Please use the format below.
```
{
  "user": {
        "email": "teamposei@user.com",
        "username": "user",
        "password": "user1234"
  }
}
``` 
Go to your email and verify your account by clicking the link that will be provided to you.
You should see a successfully verified message.

### What are the relevant Pivotal Tracker Stories
#162162999
